File: lambda.js
const AWS = require('aws-sdk');

const sns = new AWS.SNS();

const message = 'Hello, world!';

const topicArn = SNS_TOPIC_ARN;

const params = {
    Message: message,
    TopicArn: topicArn
};

sns.publish(params, (err, data) => {
    if (err) {
        console.error('Failed to publish message:', err);
    } else {
        console.log('Message published successfully:', data.MessageId);
    }
});

File: lambda.py
import os
import boto3

sns = boto3.client('sns')

message = os.environ['SNS_MESSAGE']

topic_arn = os.environ['SNS_TOPIC_ARN']

params = {
    'Message': message,
    'TopicArn': topic_arn
}

def main():
    response = sns.publish(**params)

    if 'MessageId' in response:
        print('Message published successfully:', response['MessageId'])
    else:
        print('Failed to publish message:', response['ResponseMetadata']['HTTPStatusCode'])
